client/state/data-layer/wpcom/account-recovery/lookup/index.js

```diff
@@ -0,0 +1,29 @@
+/**
+ * Internal dependencies
+ */
+import wpcom from 'lib/wp';
+
+import {
+	ACCOUNT_RECOVERY_RESET_OPTIONS_REQUEST,
+	ACCOUNT_RECOVERY_RESET_OPTIONS_RECEIVE,
+	ACCOUNT_RECOVERY_RESET_OPTIONS_ERROR,
+} from 'state/action-types';
+
+export const requestResetOptions = ( { dispatch }, { userData } ) => (
+	wpcom.req.get( {
+		body: userData,
+		apiNamespace: 'wpcom/v2',
+		path: '/account-recovery/lookup',
+	} ).then( options => dispatch( {
+		type: ACCOUNT_RECOVERY_RESET_OPTIONS_RECEIVE,
+		options,
```

Comment:
Agree. To be honest, I don't have a clear answer to the fine-grained level we need or what we should do to invalidate data. In case of this PR, maybe it is enough to keep it simple as it is now. We might have clearer picture of these questions after we moved more data flow into this layer, since that's often when patterns and requirements emerge by themselves :)

How do you think?